control-plane/api/v1alpha1/api_gateway_types.go

```diff
@@ -54,6 +54,9 @@ type GatewayClassConfigSpec struct {
 	// Deployment defines the deployment configuration for the gateway.
 	DeploymentSpec DeploymentSpec `json:"deployment,omitempty"`
 
+	// Resources defines the resource requirements for the gateway.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
It seems to me like it might make sense to nest in the `DeploymentSpec`. Thoughts?

Reply:
I like that! I'll go do that.